packages/tests-unit/tests/core/routing/cacheInterceptor.test.ts

```diff
@@ -0,0 +1,238 @@
+  name: "mock",
+  get: vi.fn(),
+  set: vi.fn(),
+  delete: vi.fn(),
+};
+
+const tagCache = {
+  name: "mock",
+  getByTag: vi.fn(),
+  getByPath: vi.fn(),
+  getLastModified: vi.fn(),
+  writeTags: vi.fn(),
+};
+
+const queue = {
+  name: "mock",
+  send: vi.fn(),
+};
+
+globalThis.incrementalCache = incrementalCache;
+globalThis.tagCache = tagCache;
+
+declare global {
+  var queue: Queue;
+}
+globalThis.queue = queue;
+
+beforeEach(() => {
+  vi.clearAllMocks();
+});
+
+describe("cacheInterceptor", () => {
+  it("should take no action when next-action header is present", async () => {
+    const event = createEvent({
+      headers: {
+        "next-action": "something",
+      },
+    });
+    const result = await cacheInterceptor(event);
+
+    expect(result).toEqual(event);
+  });
+
+  it("should take no action when x-prerender-revalidate header is present", async () => {
+    const event = createEvent({
+      headers: {
+        "x-prerender-revalidate": "1",
+      },
+    });
+    const result = await cacheInterceptor(event);
+
+    expect(result).toEqual(event);
+  });
+
+  it("should take no action when incremental cache throws", async () => {
+    const event = createEvent({
+      url: "/albums",
+    });
+
+    incrementalCache.get.mockRejectedValueOnce(new Error("mock error"));
+    const result = await cacheInterceptor(event);
+
+    expect(result).toEqual(event);
+  });
+
+  it("should retrieve app router content from cache", async () => {
+    const event = createEvent({
+      url: "/albums",
+    });
+    incrementalCache.get.mockResolvedValueOnce({
+      value: {
+        type: "app",
+        html: "Hello, world!",
+      },
+    });
+
+    const result = await cacheInterceptor(event);
+
+    const body = await fromReadableStream(result.body);
+    expect(body).toEqual("Hello, world!");
+    expect(result).toEqual(
+      expect.objectContaining({
+        type: "core",
+        statusCode: 200,
+        isBase64Encoded: false,
+        headers: expect.objectContaining({
+          "cache-control": "s-maxage=31536000, stale-while-revalidate=2592000",
+          "content-type": "text/html; charset=utf-8",
+          etag: expect.any(String),
+          "x-opennext-cache": "HIT",
+        }),
+      }),
+    );
+  });
+
+  it("should take no action when tagCache lasModified is -1", async () => {
+    const event = createEvent({
+      url: "/albums",
+    });
+    incrementalCache.get.mockResolvedValueOnce({
+      value: {
+        type: "app",
+        html: "Hello, world!",
+      },
+    });
+    tagCache.getLastModified.mockResolvedValueOnce(-1);
+
+    const result = await cacheInterceptor(event);
+
+    expect(result).toEqual(event);
+  });
+
+  it("should retrieve page router content from cache", async () => {
+    const event = createEvent({
+      url: "/revalidate",
+    });
+    incrementalCache.get.mockResolvedValueOnce({
+      value: {
+        type: "page",
+        html: "Hello, world!",
+      },
```

Comment:
Since this route is ISR, we also want to test other case like when the page is a HIT.
You'll need to add `lastModified` here for it to work, and change the expected result as well.
I'm fine with merging this right now and we'll add this later in a subsequent PR if you want, in this case just add a TODO here